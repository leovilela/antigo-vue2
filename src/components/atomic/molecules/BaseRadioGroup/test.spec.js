import { shallowMount } from "@vue/test-utils"
import LRadioGroup from "./index"
let wrapper

describe("Component BaseRadioGroup <base-radio-group />", () => {
  beforeEach(() => {
    wrapper = shallowMount(LRadioGroup, {
      stubs: {
        'Base-Radio': '<div class="base-radio" />'
      },
      propsData: {
        radios: [
          { value: "1", label: "1" },
          { value: "2", label: "2" },
          { value: "3", label: "3" }
        ]
      }
    })
  })

  afterEach(() => {
    wrapper.destroy();
  })

  it("is called", () => {
    expect(wrapper.name()).toBe("LRadioGroup")
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("has exact same number of elements of radios prop", () => {
    const radiosWrapper = wrapper.findAll(".base-radio")
    expect(wrapper.vm.radios).toHaveLength(3)
    expect(radiosWrapper).toHaveLength(3)
  })
})
