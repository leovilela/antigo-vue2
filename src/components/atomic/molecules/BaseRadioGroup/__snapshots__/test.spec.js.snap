// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BaseRadioGroup <base-radio-group /> render correctly 1`] = `
<div class="base-radio-group">
  <div class="base-radio" nativevalue="1"></div>
  <div class="base-radio" nativevalue="2"></div>
  <div class="base-radio" nativevalue="3"></div>
</div>
`;
